### PR TITLE
[SMBC] New object type Business Configuration Maintenance Objects

### PR DESCRIPTION
**Pull request by huber-nicolas**
We should put an AFF Example Object for SMBC into package TEST_AFF_EXAMPLES

### CODE REVIEW

File: file-formats/smbc/examples/z_aff_example_smbc.smbc.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "formatVersion": "1",
+  "header": {
+    "description": "AFF Example Object for SMBC",
+    "originalLanguage": "en",
+    "abapLanguageVersion": "standard"
```
Comment (schneidermic0) [inline file-formats/smbc/examples/z_aff_example_smbc.smbc.json:6]: `"abapLanguageVersion": "standard"` shouldn't be generated, should it?

File: file-formats/smbc/README.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# SMBC File Format
+This file format is used for the ADT Editor for SMBC
```
[the remark this answers is not in the document]
Comment (huber-nicolas) [reply file-formats/smbc/README.md:2]: thanks. I am going to change it in all readme I uploaded

File: file-formats/smbc/type/zif_aff_smbc_v1.intf.abap [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+      "! Description of the Business Configuration shown in the Maintain Business Configurations app.
+      "! Can be translated with the Maintain Translations app
+      "! $required
+      description           TYPE c LENGTH 50,
+      "! <p class="shorttext">Original Language</p>
+      "! Original language of the development object.
+      "! Attributes Name and Description must be maintained in this language
+      "! $required
+      original_language     TYPE sy-langu,
+      "! $required
```
[the remark this answers is not in the document]
Comment (wurzka) [reply file-formats/smbc/type/zif_aff_smbc_v1.intf.abap:188]: Yes, it is. There is also no default specified. As "standard" has value space, it is not important in this case. But I would suggest to add a default as it is done for all the other headers

[the remark this answers is not in the document]
Comment (huber-nicolas) [reply file-formats/smbc/type/zif_aff_smbc_v1.intf.abap:188]: yes. $required is the problem.
required fields are always rendered into the json.

I would also add the annotation
"! $default {@link if_aff_types_v1.data:co_abap_language_version_cloud.standard}
to the field. Then standard will no longer be rendered

File: file-formats/smbc/examples/z_aff_example_smbc.smbc.json (hunk 1 above)
Comment (huber-nicolas) [reply file-formats/smbc/examples/z_aff_example_smbc.smbc.json:6]: you are right, we have to change the annotation for this field. See my comment below

File: file-formats/smbc/examples/z_aff_example_smbc.smbc.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  "serviceConfiguration": {
+    "serviceBinding": "/DMO/SB_CUSTOMER_MSSK_O4",
+    "serviceName": "/DMO/SD_CUSTOMER_MSSK",
+    "serviceVersion": "0001",
+    "rootEntity": "Customer",
+    "name": "Maintain customer",
+    "skipRootEntity": false
+  },
+  "appConfiguration": {
+    "listReport": {},
```
Comment (huber-nicolas) [inline file-formats/smbc/examples/z_aff_example_smbc.smbc.json:16]: Do we want to serialize these empty arrays?

File: file-formats/smbc/type/zif_aff_smbc_v1.intf.json [hunk 5]
```diff
@@ -0,0 +1,57 @@
+{
+  "formatVersion": "1",
+  "header": {
+    "description": "SMBC AFF Type",
+    "originalLanguage": "en"
+  },
+  "descriptions": {
+    "types": [
+      {
+        "name": "TY_CREATION_MODE",
```
Comment (huber-nicolas) [inline file-formats/smbc/type/zif_aff_smbc_v1.intf.json:10]: Why does the AFF Handler for interfaces only serialize the descriptions of types in system YI3?
@schneidermic0

Comment (schneidermic0) [reply file-formats/smbc/type/zif_aff_smbc_v1.intf.json:10]: See the ABAP Doc comments classified with `class="shorttext synchronized"`. For these a description in the metadata is created. If you use only `class="shorttext"` no description is created.

Does this answer your question?

Comment (huber-nicolas) [reply file-formats/smbc/type/zif_aff_smbc_v1.intf.json:10]: yes, thanks

File: file-formats/smbc/type/zif_aff_smbc_v1.intf.abap [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+    "! The Object Page of the Root Entity cannot be configured.
+    "! It is sufficient to maintain only those attributes that should deviate from the standard behavior.
+    BEGIN OF ty_settings,
+      "! $showAlways
+      list_report  TYPE ty_list_report,
+      "! $showAlways
+      object_pages TYPE ty_object_pages,
+    END OF ty_settings.
+  TYPES:
+*        See also {@link if_mbc_cp_api_business_config}
```
Comment (schneidermic0) [inline file-formats/smbc/type/zif_aff_smbc_v1.intf.abap:146]: Shall we remove this?
```suggestion
```

File: file-formats/smbc/examples/z_aff_example_smbc.smbc.json (hunk 4 above)
Comment (schneidermic0) [reply file-formats/smbc/examples/z_aff_example_smbc.smbc.json:16]: If possible, I would skip them by removing `$showAlways`, but the object type owners have to decide how import it is for them to list an empty `appConfiguration`

File: file-formats/smbc/type/zif_aff_smbc_v1.intf.abap [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+      "! <p class="shorttext">Service Version</p>
+      "! The version of the service to be used (as defined in the Service Binding)
+      "! $required
+      service_version  TYPE n LENGTH 4,
+      "! <p class="shorttext">Root Entity Set</p>
+      "! The name of an entity set as exposed by the Service Definition.
+      "! This entity set is used as the root node for the UI.
+      "! Only for this root entity set and its associations a UI is shown.
+      "! $required
+      root_entity      TYPE c LENGTH 30,
```
Comment (schneidermic0) [inline file-formats/smbc/type/zif_aff_smbc_v1.intf.abap:166]: I am not sure is the addition would be `_set` important? See the field's title

File: file-formats/smbc/type/zif_aff_smbc_v1.intf.abap [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+      hide_sort              TYPE ty_personalization-hide_sort,
+      "! <p class="shorttext synchronized">Hide Column</p>
+      "! If true, the user cannot add and remove columns to the table
+      hide_column            TYPE ty_personalization-hide_column,
+    END OF ty_list_report,
+    BEGIN OF ty_object_page_common,
+      "! <p class="shorttext synchronized">Entity</p>
+      "! Entity Name of the Object Page
+      entity                  TYPE string,
+      type                    TYPE ty_table_type,
```
Comment (schneidermic0) [inline file-formats/smbc/type/zif_aff_smbc_v1.intf.abap:94]: `type` or `table_type`?

**Review by schneidermic0 — COMMENTED** [review]
I have some minor comments. Maybe, the object type owner want to check them.

File: file-formats/smbc/examples/z_aff_example_smbc.smbc.json (hunk 4 above)
Comment (huber-nicolas) [reply file-formats/smbc/examples/z_aff_example_smbc.smbc.json]: $showAlways was removed

File: file-formats/smbc/type/zif_aff_smbc_v1.intf.abap (hunk 6 above)
Comment (huber-nicolas) [reply file-formats/smbc/type/zif_aff_smbc_v1.intf.abap]: the comment was deleted

File: file-formats/smbc/type/zif_aff_smbc_v1.intf.abap (hunk 7 above)
Comment (huber-nicolas) [reply file-formats/smbc/type/zif_aff_smbc_v1.intf.abap]: _set was added and the descriptions was changed

File: file-formats/smbc/type/zif_aff_smbc_v1.intf.abap (hunk 8 above)
Comment (huber-nicolas) [reply file-formats/smbc/type/zif_aff_smbc_v1.intf.abap]: changed to table_type

**Review by schneidermic0 — APPROVED** [review]
Thanks :)

LGTM

**Pull request merged**